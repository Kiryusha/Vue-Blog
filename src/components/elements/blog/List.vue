<template lang="pug">
  section.blog-list(:class="{'_loaded': list.length}")
    .blog-list__content
      .blog-list__item(
        v-for="item in list"
      )
        ListItem(
          :data="item"
        )
</template>

<script>
import ListItem from '../../elements/blog/ListItem';

export default {
  components: {
    ListItem,
  },
  props: {
    list: {
      type: Array,
      default: () => ([]),
    },
  },
  mounted() {
    if (this.list.length) {
      this.$Progress.finish();
    }
  },
};
</script>

<style lang="stylus">
  .blog-list
    card()
    opacity 0
    transition opacity .2s

    &._loaded
      opacity 1

    &__item

      & + &
        margin-top 75px

    &__title
      font-family 'Roboto Slab', sans-serif
      font-size 30px
      margin-bottom 20px

    &__link
      text-decoration none
      color $yellowDark
      border-bottom 1px solid $yellowDark
      transition .2s

      &:hover
        color $yellow
        border-color $yellow

</style>
