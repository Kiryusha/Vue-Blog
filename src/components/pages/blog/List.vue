<template lang="pug">
  section.blog-list(:class="{'_loaded': this.list.length}")
    .blog-list__content
      .blog-list__item(
        v-for="item in list"
      )
        ListItem(
          :data="item"
        )
</template>

<script>
import ListItem from '../../elements/blog/ListItem';

export default {
  components: {
    ListItem,
  },
  props: {
    list: {
      type: Array,
      default: () => ([]),
    },
  },
  mounted() {
    if (this.list.length) {
      this.$Progress.finish();
    }
  },
};
</script>

<style lang="stylus">
  .blog-list
    card()
    background transparent
    box-shadow 0 0 20px rgba(0, 0, 0, 0)

    &._loaded
      background #f
      box-shadow 0 0 20px rgba(0, 0, 0, .2)

    &__item

      & + &
        margin-top 75px
</style>
