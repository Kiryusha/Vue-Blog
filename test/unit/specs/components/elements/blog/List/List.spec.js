// List.vue

import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import List from 'Components/elements/blog/List';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('List.vue', () => {
  const endlessScroll = jest.fn();
  const config = {
    localVue,
    mocks: {
      $route: {
        params: {
          code: 'test',
        }
      }
    }
  };

  let storeConfig = {
    state: {
      list: {
        list: [{}, {}],
        currentPage: 1,
        activeCategory: '',
        isBottomReached: false,
      },
    },
    actions: {
      fetchList: () => {},
    }
  };
  let store = new Vuex.Store(storeConfig);

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('component shallow rendering', () => {
    it('should match a snapshot', () => {
      const wrapper = shallowMount(List, { ...config, store });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe("lifecycle-driven methods calls", () => {
    it('should attach endlessScroll() to the window scroll event on a created hook', async () => {
        const wrapper = shallowMount(List, {...config, store, methods: {
          endlessScroll,
        }});
        await wrapper.vm.$nextTick();
        global.window.dispatchEvent(new Event('scroll'));
        wrapper.destroy();

        expect(endlessScroll).toHaveBeenCalled();
      });

    it('should remove endlessScroll() from the window scroll event on a destroyed hook', async () => {
      const wrapper = shallowMount(List, {...config, store, methods: {
        endlessScroll,
      }})
      await wrapper.destroy();
      global.window.dispatchEvent(new Event('scroll'));

      expect(endlessScroll).not.toHaveBeenCalled();
    });
  });
});
