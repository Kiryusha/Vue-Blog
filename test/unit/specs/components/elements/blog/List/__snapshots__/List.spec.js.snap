// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue component shallow rendering should match a snapshot 1`] = `
<section class="list _loaded">
  <div class="content">
    <div class="list-item">
      <item-stub></item-stub>
    </div>
    <div class="list-item">
      <item-stub></item-stub>
    </div>
  </div>
</section>
`;
